### PR TITLE
sixel support

### LINKED ISSUE #422: Is inline image display a future possibility?
We have downloads via the print escape sequence support, but it would be nice to also have inline image display support.  Is there any possibility of this as a feature for wetty?  

Thanks,  
Marc

### PR DESCRIPTION
**Pull request by jerch**
May fix #422.

@butlerx Well plz see the comments in source file - since I dont quite grasped your build layout I simply added the worker file to assets, which is prolly not the way to do it.

### CODE REVIEW

File: src/client/wetty/term.ts
Comment (butlerx) [inline src/client/wetty/term.ts:26]: I wonder if the file could be pulled In from web_modules but this definite works given the add on is still in testing

**Review by butlerx — APPROVED** [review]

**Comment by butlerx** [comment]
You'll need to add the worker file to the prettier and eslint ignore to pass tests

File: src/client/wetty/term.ts
Comment (jerch) [reply src/client/wetty/term.ts:26]: Well didnt quite figure out how web_modules gets composed or pulls things from packages, thats why the assets hack. :smile: 

How could it be done directly via web_modules?

Comment (jerch) [reply src/client/wetty/term.ts:26]: Btw the sixel path is pretty consolidated (in beta), the alpha/experimental is still on the module due other pending image support (at least png and jpg is planned).
For wetty the worker approach is def. a nuisance (due to having only one terminal in a document), there is an issue open to also provide a non workered version (no clue yet how to shape that neatly).

Comment (butlerx) [reply src/client/wetty/term.ts:26]: For reference the web_modules is built by snowpack it parses out the node_modules imported by the clients moves them to the web_modules folder and updates the import path.
The server then just serves them and I leave it up to the browsers to handle the imports and caching of js files.

Comment (jerch) [reply src/client/wetty/term.ts:26]: Thx for the explanation. I messed around with snowpack for a while and tried to get working in a more straight forward way:
- packing the worker file in separate package `xterm-addon-image-worker` does not work, as snowpack decorates the worker with an import declaration, which is not yet supported by all browser for workers, maybe there is a special worker flag to avoid that (did not find any)
- generate a `worker.ts` in wetty project and import image worker there hoping it will handle it all right - nope, same import error as above

Finally this works:
- code as `this.loadAddon(new ImageAddon('/wetty/web_modules/pkg/xterm-addon-image-worker.js'));`
- yarn build
- cp node_modules/xterm-addon-image/lib/xterm-addon-image-worker.js.map build/web_modules/pkg/
- yarn start

Again this manual copy is annoying, maybe this could be integrated into snowpack build flow easier. Not sure if thats any better than treating the worker file as a static js asset.

(Sidenote: Proper bundling of worker stuff is really messed up in almost all buildchain tools. Seems thats not a an often usecase at all.)

Comment (butlerx) [reply src/client/wetty/term.ts:26]: Thanks for all that. Snowpack can just do moves and regular commands. But I think for this case the vendor ing of the file its fine to get the feature rolled out.

If we need to update the file in the future we can re look at how to manually add the dependency. If all else fails moving back to bash will always work.

Comment (jerch) [reply src/client/wetty/term.ts:26]: So moving it "hard" to assets (and exclude it from linter) will do for now?

Comment (butlerx) [reply src/client/wetty/term.ts:26]: Yep

**Comment by jerch** [comment]
@butlerx Reverted to the asset copy solution and added ingore files for eslint and prettier. This is still subpar, as it involves manual interaction once the package version was changed (file needs to be copied over again).

Seems the tests wont run without your approval?

**Comment by butlerx** [comment]
Thanks @jerch

**Pull request merged**
